ethereum/geth_client.go

```diff
@@ -347,11 +347,12 @@ func (gc *gethClient) SubmitTransactionWithRetries(contractMethod interface{}, o
 		}
 
 		if err == nil {
+			log.Infof("TX %s successfully sent with nonce %s", tx.Hash().String(), opts.Nonce.String())
 			return tx, nil
 		}
 
-		if (err.Error() == ErrTransactionUnderpriced.Error()) || (err.Error() == ErrNonceTooLow.Error()) {
-			log.Warningf("Concurrent transaction identified, trying again [%d/%d]\n", current, maxTries)
+		if strings.Contains(err.Error(), ErrIncNonce.Error()) || strings.Contains(err.Error(), ErrIncPrice.Error()) {
```

Comment:
Fixing this will not ensure that all txs will always go through, since they can still fail silently in the processing eth node.